apps/condo/domains/billing/schema/ValidateQRCodeService.js

```diff
@@ -0,0 +1,107 @@
+ */
+const { isEmpty } = require('lodash')
+
+const { GQLError, GQLErrorCode: { BAD_USER_INPUT, INTERNAL_ERROR } } = require('@open-condo/keystone/errors')
+const { GQLCustomSchema } = require('@open-condo/keystone/schema')
+
+const { AcquiringIntegrationContext } = require('@condo/domains/acquiring/utils/serverSchema')
+const { BankAccount } = require('@condo/domains/banking/utils/serverSchema')
+const access = require('@condo/domains/billing/access/ValidateQRCodeService')
+const { WRONG_FORMAT, NOT_FOUND } = require('@condo/domains/common/constants/errors')
+const { Organization } = require('@condo/domains/organization/utils/serverSchema')
+
+/**
+ * List of possible errors, that this custom schema can throw
+ * They will be rendered in documentation section in GraphiQL for this custom schema
+ */
+const ERRORS = {
+    INVALID_QR_CODE: {
+        mutation: 'validateQRCode',
+        variable: ['data', 'qrCode'],
+        code: BAD_USER_INPUT,
+        type: WRONG_FORMAT,
+        message: 'Provided QR code does\'nt have one of required fields: PersonalAcc, payerAddress, lastName, paymPeriod or Sum',
+    },
+    NO_ORGANIZATION: {
+        mutation: 'validateQRCode',
+        code: INTERNAL_ERROR,
+        type: NOT_FOUND,
+        message: 'Organization with provided TIN is not registered with Doma.ai',
+    },
+    NO_ACQUIRING_CONTEXT: {
+        mutation: 'validateQRCode',
+        code: INTERNAL_ERROR,
+        type: NOT_FOUND,
+        message: 'Organization with provided TIN does not have an active AcquiringIntegrationContext',
+    },
+    NO_BANK_ACCOUNT: {
+        mutation: 'validateQRCode',
+        code: INTERNAL_ERROR,
+        type: NOT_FOUND,
+        message: 'Bank Account with provided personal account number wasn\'t found',
+    },
+}
+
+const ValidateQRCodeService = new GQLCustomSchema('ValidateQRCodeService', {
+    types: [
+        {
+            access: true,
+            type: 'input ValidateQRCodeInput { dv: Int!, sender: SenderFieldInput!, qrCode: String! }',
+        },
+        {
+            access: true,
+            type: 'type ValidateQRCodeOutput { qrCodeFields: JSON! }',
+        },
+    ],
+    
+    mutations: [
+        {
+            access: access.canValidateQRCode,
+            schema: 'validateQRCode(data: ValidateQRCodeInput!): ValidateQRCodeOutput',
+            resolver: async (parent, args, context) => {
+                const { data } = args
+                const { qrCode } = data
+
+                const qrCodeFields = Object.fromEntries(qrCode.split('|').map(part => part.split('=')))
+                const { PersonalAcc, payerAddress, lastName, paymPeriod, Sum, PayeeINN } = qrCodeFields
+
+                if (!PersonalAcc || !payerAddress || !lastName || !paymPeriod || !Sum) throw new GQLError(ERRORS.INVALID_QR_CODE)
+                
+                if (PayeeINN) {
+                    const organizations = await Organization.getAll(context, {
```

Comment:
Is there only one organization with given TIN?
```suggestion
                    const organization = await Organization.getOne(context, {
```

Reply:
to be on the safe side we only use getOne when querying by id, otherwise if more than one object is found it's gonna crash